akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala

```diff
@@ -109,7 +120,9 @@ object Sink {
    * If `f` throws an exception and the supervision decision is
    * [[akka.stream.Supervision.Resume]] or [[akka.stream.Supervision.Restart]] the
    * element is dropped and the stream continues.
+   * @deprecated Use `foreachAsync` instead, it allows you to choose your own execution context.
    */
+  @deprecated("Use `foreachAsync` instead, it allows you to choose your own execution context.", since = "2.5.13")
```

Comment:
"it allows you to choose your own execution context"

well, that was the case for `foreachParallel` also, since it's a parameter here

"it allows you to choose how to run the procedure, by calling some other API returning a CompletionStage or using CompletableFuture.supplyAsync"